app/javascript/packs/form-steps-wait.js

```diff
@@ -0,0 +1,76 @@
+import { loadPolyfills } from '@18f/identity-polyfill';
+
+/**
+ * @typedef FormStepsWaitElements
+ *
+ * @prop {HTMLFormElement} form
+ */
+
+/**
+ * @typedef FormStepsWaitOptions
+ *
+ * @prop {number} pollIntervalMs Poll interval.
+ * @prop {string} waitStepPath URL path to wait step, used in polling.
+ */
+
+/** @type {FormStepsWaitOptions} */
+const DEFAULT_OPTIONS = {
+  pollIntervalMs: 3000,
+  waitStepPath: `${window.location.pathname}_wait`,
+};
+
+class FormStepsWait {
+  constructor(form) {
+    /** @type {FormStepsWaitElements} */
+    this.elements = { form };
+
+    this.options = {
+      ...DEFAULT_OPTIONS,
+      ...this.elements.form.dataset,
+    };
+
+    this.options.pollIntervalMs = Number(this.options.pollIntervalMs);
+
+    this.bindEvents();
+  }
+
+  bindEvents() {
+    this.elements.form.addEventListener('submit', (event) => this.handleSubmit(event));
+  }
+
+  /**
+   * @param {Event} event Form submit event.
+   */
+  async handleSubmit(event) {
+    event.preventDefault();
+
+    const { form } = this.elements;
+    const { action, method } = form;
+
+    await window.fetch(action, {
+      method,
+      body: new window.FormData(form),
+      redirect: 'manual',
+    });
```

Comment:
Should probably handle a failure here. Reload the page? Off the top of my head, not sure if/when we'd ever expect it to fail.

Reply:
> Should probably handle a failure here. Reload the page (to show flash messages)? Off the top of my head, not sure if/when we'd ever expect it to fail.

Handling as part of consistent response handler in 5e6b201.